references/detection/train_tpu.py

```diff
@@ -128,28 +149,23 @@ def process_prediction(output, threshold=0.05):
     return masks, pred_boxes, pred_class
 
 
-def do_prediction(image_path, use_cpu=False):
-    model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True)
-    model.eval()
-
+def do_prediction(image_path, model, use_cpu=False):
     image = Image.open(image_path)
-    image = torchvision.transforms.functional.resize(image, (800, 600))
     image_tensor = torchvision.transforms.functional.to_tensor(image)
 
-    if not use_cpu:
-        # Send to single TPU device
-        device = xm.xla_device()
-        torch_xla._XLAC._xla_set_default_device(str(device))
-        model = model.to(device)
-        image_tensor = image_tensor.to(device)
+    with _bench('inference'):
+        output = model([image_tensor])
+        torch_xla._XLAC._xla_sync_multi(list(output[0].values()), devices=[])
 
-    output = model([image_tensor])
-    print("output: {}".format(output))
-    print(torch_xla._XLAC._xla_metrics_report())
+    print('# of boxes:', len(output[0]['boxes']))
+
+    with _bench('cpu postprocessing'):
+        masks, pred_boxes, pred_class = process_prediction(output)
+        image = overlay_output(image, masks, pred_boxes, pred_class)
+        image.save(os.path.join('/coco/out', os.path.basename(image_path)))
 
-    masks, pred_boxes, pred_class = process_prediction(output)
-    image = overlay_output(image, masks, pred_boxes, pred_class)
-    image.save(os.path.join('/coco/out', os.path.basename(image_path)))
+    print()
+    print(torch_xla._XLAC._xla_metrics_report())
```

Comment:
when we run this multi-core or pods, this would print a lot of things to the screen. Instead, let's use `xm.master_print`